third_party/proton/csrc/lib/Driver/GPU/HipApi.cpp

```diff
@@ -37,13 +40,20 @@ Device getDevice(uint64_t index) {
   (void)hip::deviceGetAttribute<true>(
       &smCount, hipDeviceAttributeMultiprocessorCount, index);
 
-  // TODO: Compute capability is a NVIDIA concept. It doesn't map naturally to
-  // AMD GPUs. Figure out a better way to support this.
-  uint64_t arch = 0;
+  std::string arch = getHipArchName(index);
+
   return Device(DeviceType::HIP, index, clockRate, memoryClockRate, busWidth,
                 smCount, arch);
 }
 
+const std::string getHipArchName(uint64_t index) {
+  hipDeviceProp_t devProp;
```

Comment:
There is a bug regarding this so I'm not sure `hipDeviceProp_t` + `getDeviceProperties` work as expected when dynamic loaded (which is the case in triton). See https://github.com/triton-lang/triton/blob/main/third_party/amd/backend/driver.c#L22 for details. That's why we are using `deviceGetAttribute` in the above..

Reply:
OK. Updated the calls to hipGetDevicePropertiesR0600. We should make a point to go back and switch these out when the new Rocm updates land.

Reply:
We cannot use `hipGetDevicePropertiesR0600`--that symbol does not exist on ROCm 5.x and PyTorch need to support 5.x still.. So need to use the deprecated `hipDeviceProp_tR0000`. (Yeah we created a mess for ourselves there in ROCm stack..)

Though I recall that @mwootton runs into crashes when calling `getDeviceProperties`. Does everything work fine on your side?

Reply:
Interesting. Right, the tests pass on both the CI and locally for me so I wasn't sure. hipDeviceAttribute_t doesn't support hipDeviceAttributeGcnArch anymore so I think we need to use hipGetDeviceProperties no matter what.

Reply:
Sorry I wasn't being clear. Yes `getDeviceProperties` + the deprecated `hipDeviceProp_tR0000` struct works fine in the main triton compiler.

Reply:
Synced offline. This works as the moment so I'll unblock this. But will keep investigating a bit into it and make the story more robust.